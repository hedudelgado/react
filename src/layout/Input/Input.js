import React from 'react';
import {
  oneOf,
  string,
  bool,
  func,
  object,
} from 'prop-types';
import classnames from 'classnames';

const Input = ({
  field,
  sessioncamhide,
  touched,
  error,
  id,
  label,
  mb,
  ...props
}) => {
  const showError = error && touched;
  const showValid = !error && touched;
  const cssClasses = classnames({
    sessioncamhidetext: sessioncamhide,
    'error-field': showError,
  }, 'wb-dark');
  const parentCssClasses = `wb-form-item ${mb}`;

  return (
    <div className={parentCssClasses}>
      {label && <label className="wb-strong" htmlFor={id}>{label}</label>}
      <div className="wb-form-item__field wb-form-item__field__full">
        <input
          className={cssClasses}
          id={id}
          label={label}
          {...field}
          {...props}
        />
        {showError && <span className="form-item-msg__error-icon wb-icon_disc-info-filled" />}
        {showValid && <span className="form-item-msg__info-icon wb-icon_disc-checked-fill" />}
      </div>
      {showError && <span className="form-item-msg form-item-msg--error">{error}</span>}
    </div>
  );
};

Input.propTypes = {
  /**
    HTML type attribute.
  */
  type: oneOf([
    'text',
    'email',
    'tel',
    'password',
  ]),
  id: string,
  /**
    HTML placeholder attribute.
  */
  placeholder: string,
  /**
    HTML value attribute.
  */
  value: string,
  /**
    Input label. Not displayed if empty.
  */
  label: string,
  /**
    Sessioncam field masking.
  */
  sessioncamhide: bool,
  /**
    Error message. Not displayed if empty.
  */
  error: string,
  /**
    Function to be run on input change.
  */
  onChange: func,
  field: object,
  touched: bool,
  mb: string,
};

Input.defaultProps = {
  type: 'text',
  mb: 'wb-push-middle--bottom',
  sessioncamhide: true,
};

export default Input;
