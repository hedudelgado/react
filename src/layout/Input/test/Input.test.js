import React from 'react';
import { shallow } from 'enzyme';

import Input from '../Input';

describe('Input Component', () => {
  const props = {
    id: 'test',
    type: 'text',
    placeholder: 'Placeholder',
    value: 'value',
    label: 'Label',
    touched: true,
  };

  it('should render the component', () => {
    const wrapper = shallow(<Input {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render label if not set', () => {
    const wrapper = shallow(<Input {...{ ...props, label: null }} />);
    expect(wrapper).not.toContainMatchingElement('label');
  });

  it('should render the error if set', () => {
    const wrapper = shallow(<Input {...{ ...props, error: 'error message' }} />);
    expect(wrapper).toContainMatchingElement('.error-field');
    expect(wrapper).toContainMatchingElement('.form-item-msg--error');
    expect(wrapper.find('.form-item-msg--error')).toHaveText('error message');
  });

  it('should set sessioncamhide class if set', () => {
    const wrapper = shallow(<Input {...{ ...props, sessioncamhide: true }} />);
    expect(wrapper.find('input')).toHaveClassName('sessioncamhidetext');
  });
});
