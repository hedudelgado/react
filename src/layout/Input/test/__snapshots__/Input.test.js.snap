// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should render the component 1`] = `
<div
  className="wb-form-item wb-push-middle--bottom"
>
  <label
    className="wb-strong"
    htmlFor="test"
  >
    Label
  </label>
  <div
    className="wb-form-item__field wb-form-item__field__full"
  >
    <input
      className="sessioncamhidetext wb-dark"
      id="test"
      label="Label"
      placeholder="Placeholder"
      type="text"
      value="value"
    />
    <span
      className="form-item-msg__info-icon wb-icon_disc-checked-fill"
    />
  </div>
</div>
`;
